src/test/java/org/truemail/ClientTests.java

```diff
@@ -0,0 +1,70 @@
+package org.truemail;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.truemail.client.Configuration;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import static junit.framework.Assert.assertEquals;
+import static org.powermock.api.mockito.PowerMockito.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Client.class})
+public class ClientTests {
+    @Test
+    public void validateReturnsSuccessResponseWhenValidEmail() throws Exception {
+        URL u = mock(URL.class);
+        String url = "https://test.host:12345?email=test.email@com";
+        whenNew(URL.class).withArguments(url).thenReturn(u);
+        HttpURLConnection huc = mock(HttpURLConnection.class);
+        when(u.openConnection()).thenReturn(huc);
+        when(huc.getResponseCode()).thenReturn(HttpURLConnection.HTTP_OK);
+        when(huc.getResponseMessage()).thenReturn("Be Nice");
+
+        Configuration config = new Configuration(true, "test.host", 12345,  "2000");
+        Client client = new Client(config);
+        String response = client.validate("test.email@com");
+
+        assertEquals("Be Nice", response);
+    }
+
+    @Test
+    public void validateReturnsErrorResponseWhenHttpConnectionFailed() throws Exception {
+        URL u = mock(URL.class);
+        String url = "https://test.host:12345?email=test.email";
+        whenNew(URL.class).withArguments(url).thenReturn(u);
+        HttpURLConnection huc = mock(HttpURLConnection.class);
```

Comment:
Needless, in this case you don't huc var.

Reply:
Good catch, interesting why IDE does not highlight this for me....

Reply:
Good catch, thanks!)